### PR TITLE
NMS-14109: Grafana box renders raw JS when Grafana behind reverse proxy with SSO

### PR DESCRIPTION
**Pull request by aramosvizcarra**
Added extra set of properties that will only be used to generate the links url to access grafana datasources. Will use original api settings if they are not declared or blank.

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14109

### CODE REVIEW

File: docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc [hunk 1]
```diff
@@ -72,6 +72,26 @@ When no tag is given all dashboards will be displayed
 | Maximum number of entries to be displayed (0 for unlimited)
 | Integer
 | 0
+
```
Comment (jeffgdotorg) [inline docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc:75]: Please update the screenshot with the attached one
Screenshot 2022-04-29 at 11-36-02 OpenNMS Web Console
.

**Review by jeffgdotorg — CHANGES_REQUESTED** [review]
Just some doc cleanup, otherwise this looks great and worked in my testing.

File: docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc [hunk 2]
```diff
@@ -72,6 +72,26 @@ When no tag is given all dashboards will be displayed
...
+| org.opennms.grafanaBox.link.protocol
+| The protocol used to build each dashboard link url.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc:77]:
```suggestion
| The protocol used to build each dashboard link URL.
```

File: docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc [hunk 3]
```diff
@@ -72,6 +72,26 @@ When no tag is given all dashboards will be displayed
...
+| String
+| org.opennms.grafanaBox.protocol
+
+| org.opennms.grafanaBox.link.hostname
+| Hostname used to build each dashboard link url.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc:82]:
```suggestion
| Hostname used to build each dashboard link URL.
```

File: docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc [hunk 4]
```diff
@@ -72,6 +72,26 @@ When no tag is given all dashboards will be displayed
...
+| String
+| org.opennms.grafanaBox.hostname
+
+| org.opennms.grafanaBox.link.port
+| The port used to build each dashboard link url.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc:87]:
```suggestion
| The port used to build each dashboard link URL.
```

File: docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc [hunk 5]
```diff
@@ -72,6 +72,26 @@ When no tag is given all dashboards will be displayed
...
+| Integer
+| org.opennms.grafanaBox.port
+
+| org.opennms.grafanaBox.link.basePath
+| The Grafana base path used to build each dashboard link url.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/admin/webui/grafana-dashboard-box.adoc:92]:
```suggestion
| The Grafana base path used to build each dashboard link URL.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
additional minor changes to the docs.

**Comment by aramosvizcarra** [comment]
> Should this be rebased to `foundation-2022` or does the SSO stuff not affect earlier releases?

I think so, @jeffgdotorg  probably know for sure.

**Review by Bonrob2 — DISMISSED** [review]
Docs LGTM!

**Comment by aramosvizcarra** [comment]
Created a new PR since I had to rebase to foundation-2022
https://github.com/OpenNMS/opennms/pull/4654

**Pull request closed without merging**